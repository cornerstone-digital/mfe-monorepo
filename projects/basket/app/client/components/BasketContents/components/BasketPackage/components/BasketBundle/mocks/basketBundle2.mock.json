{
  "basketId": "e5be8241-7b99-4c48-b2b4-d98a2eaec88a",
  "affiliateId": null,
  "affiliateFlag": false,
  "source": "WCS",
  "customerRequestedDate": "2018-11-22T13:11:00.000Z",
  "metadata": {
    "returnURL": null,
    "basketStatus": "INCOMPLETE",
    "salesOrderId": null,
    "reason": null
  },
  "customer": null,
  "journey": null,
  "packages": [{
    "packageId": "684b793c-88b6-4b3f-a4ab-863e26e37f75",
    "accountCategory": "Consumer",
    "sequence": 2,
    "packageType": "Acquisition",
    "removable": true,
    "dependentOn": [],
    "voucherCode": null,
    "bundle": {
      "packageLineId": "94ffbac8-1703-4226-94a8-d5447f296ca7",
      "action": "ADD",
      "skuId": "110546",
      "name": "Gigafast 100 with Voice -18 months contract Unlimited Broadband",
      "description": null,
      "displayName": "Vodafone Gigafast Broadband 100",
      "displayDescription": "Vodafone Gigafast Broadband 100",
      "bundleClass": "HANDSET",
      "productClass": null,
      "serviceId": "107551",
      "paymentType": "Postpaid",
      "bundleType": null,
      "commitmentPeriod": {
        "uom": "Months",
        "value": "18"
      },
      "displayOrder": null,
      "showInSummary": false,
      "allowances": null,
      "merchandisingMedia": [],
      "attributes": null,
      "priceDetails": {
        "monthlyPrice": {
          "gross": "28",
          "net": "23.33",
          "vat": "4.67"
        },
        "monthlyDiscountPrice": {
          "gross": "28",
          "net": "23.33",
          "vat": "4.67"
        },
        "merchandisingPromotions": null
      },
      "isVodafoneGlobalRoaming": false,
      "bundledServiceProducts": [{
        "skuId": "107552",
        "name": "Fixed Broadband Service",
        "description": "Fixed Broadband Service",
        "productClass": "Fixed Broadband Service",
        "servicePointId": "110560",
        "priceDetail": {
          "oneOffPrice": null,
          "oneOffDiscountPrice": null,
          "monthlyPrice": null,
          "monthlyDiscountPrice": null,
          "merchandisingPromotions": null
        },
        "miscellaneous": [{
          "name": "SERVICE POINT",
          "value": "Gigafast 100 Broadband"
        }]
      }, {
        "skuId": "107551",
        "name": "Fixed Line Service",
        "description": "Fixed Line Service",
        "productClass": "Fixed Line",
        "servicePointId": "110567",
        "priceDetail": {
          "oneOffPrice": null,
          "oneOffDiscountPrice": null,
          "monthlyPrice": null,
          "monthlyDiscountPrice": null,
          "merchandisingPromotions": null
        },
        "miscellaneous": [{
          "name": "SERVICE POINT",
          "value": "Gigafast Home Broadband and Voice"
        }]
      }],
      "planCategory": "CONSUMER",
      "isChanged": false
    },
    "hardwares": [{
      "packageLineId": "06b180fe-cc4c-4d2b-b4a5-e283ddcbced7",
      "action": "ADD",
      "skuId": "200692",
      "name": "Vodafone VOX 3.0 Uk",
      "description": null,
      "displayName": "Vodafone Broadband Router",
      "displayDescription": null,
      "deviceBriefDescription": [],
      "deviceColor": null,
      "memorySize": null,
      "make": null,
      "model": null,
      "groupType": null,
      "productClass": "Fixed Broadband Equipment",
      "displayOrder": null,
      "showInSummary": false,
      "merchandisingMedia": [{
        "id": "imageURLs.thumbs.front",
        "value": "/assets/images/desktop/IMG_170x170_VOIX_30_Router.png",
        "type": "URL",
        "footNotes": [""]
      }],
      "attributes": [],
      "priceDetails": {
        "oneOffPrice": {
          "gross": "0",
          "net": "0",
          "vat": "0"
        },
        "oneOffDiscountPrice": {
          "gross": "0",
          "net": "0",
          "vat": "0"
        },
        "merchandisingPromotions": null,
        "monthlyPrice": null,
        "monthlyDiscountPrice": null,
        "financingOptions": [],
        "deviceFinancingId": null
      },
      "boxPrice": {
        "gross": "69.72",
        "net": "58.10",
        "vat": "11.62"
      },
      "freeGift": false,
      "isChanged": false
    }],
    "services": [{
      "packageLineId": "6572d967-0aae-4cad-a032-e65860ec988b",
      "action": "ADD",
      "skuId": "110565",
      "name": "Standard Installation Fee",
      "description": null,
      "displayName": "Standard Installation Fee",
      "displayDescription": null,
      "specificationGroup": [],
      "merchandisingMedia": [],
      "attributes": [],
      "priceDetails": {
        "oneOffPrice": {
          "gross": "0",
          "net": "0",
          "vat": "0"
        },
        "oneOffDiscountPrice": {
          "gross": "0",
          "net": "0",
          "vat": "0"
        },
        "monthlyPrice": null,
        "monthlyDiscountPrice": null,
        "merchandisingPromotions": null
      },
      "freeGift": false,
      "productClass": "Installation Fee",
      "dataRollover": null,
      "rootProductId": null,
      "allowances": null,
      "rewardPoints": null,
      "planDuration": null,
      "isChanged": false
    }, {
      "packageLineId": "6572d967-0aae-4cad-a032-e65860ec988b",
      "action": "ADD",
      "skuId": "110565",
      "name": "Standard",
      "description": null,
      "displayName": "Standard",
      "displayDescription": "Standard",
      "specificationGroup": [],
      "merchandisingMedia": [],
      "attributes": [],
      "priceDetails": {
        "oneOffPrice": {
          "gross": "0",
          "net": "0",
          "vat": "0"
        },
        "oneOffDiscountPrice": {
          "gross": "0",
          "net": "0",
          "vat": "0"
        },
        "monthlyPrice": null,
        "monthlyDiscountPrice": null,
        "merchandisingPromotions": null
      },
      "freeGift": false,
      "productClass": "Installation Fee",
      "dataRollover": null,
      "rootProductId": null,
      "allowances": null,
      "rewardPoints": null,
      "planDuration": null,
      "isChanged": false
    }],
    "discounts": [],
    "priceDetails": {
      "oneOffPrice": {
        "gross": "0",
        "net": "0",
        "vat": "0"
      },
      "oneOffDiscountPrice": {
        "gross": "0",
        "net": "0",
        "vat": "0"
      },
      "monthlyPrice": {
        "gross": "28",
        "net": "23.33",
        "vat": "4.67"
      },
      "monthlyDiscountPrice": {
        "gross": "28",
        "net": "23.33",
        "vat": "4.67"
      },
      "financingOptions": []
    },
    "installationAddress": {
      "citySubDivisionName": null,
      "country": "United Kingdom",
      "county": null,
      "flatNumber": null,
      "houseName": null,
      "houseNumber": "89",
      "locality": null,
      "moveTypeCode": "",
      "postCode": "YO30 5TB",
      "streetName": "EASTHOLME DRIVE",
      "town": "York"
    },
    "appointmentWindow": null,
    "servicePoint": null,
    "serviceStartDate": null,
    "phoneNumber": null,
    "linePackageType": null,
    "lineTreatmentType": null,
    "planType": "BroadBand:FTTH",
    "isChanged": false
  }],
  "vetOutcome": null,
  "priceDetails": {
    "oneOffPrice": {
      "gross": "0",
      "net": "0",
      "vat": "0"
    },
    "oneOffDiscountPrice": {
      "gross": "0",
      "net": "0",
      "vat": "0"
    },
    "monthlyPrice": {
      "gross": "28",
      "net": "23.33",
      "vat": "4.67"
    },
    "monthlyDiscountPrice": {
      "gross": "0",
      "net": "0",
      "vat": "0"
    }
  },
  "totalOneoffPriceSaving": "0.0",
  "totalMonthlyPriceSaving": "84",
  "deliveryInfo": null,
  "voucherCode": null,
  "voucherDescription": null,
  "validated": false,
  "hashvalue": "1003091324",
  "changeCode": null,
  "voucherErrorMessage": null,
  "activeBundles": [],
  "isChanged": false
}
