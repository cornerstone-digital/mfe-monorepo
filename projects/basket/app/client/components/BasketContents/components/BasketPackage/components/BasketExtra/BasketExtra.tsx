import containsText from '@web-shop-core/helpers/containsText'

import BasketItem from '@components/BasketItem'

import getPrice from '@helpers/getPrice'
import getBundleDescription from '../../helpers/getBundleDescription'

import { BasketExtraProps } from './BasketExtras.types'
import { isBroadband, isFixedLineService, isInsuranceService } from '@helpers/typeCheck'
import { ServiceValues } from '../PackageItemList/PackageItemList.types'
import isHiddenService from './helpers/isHiddenService'
import getPicture from './helpers/getPicture'

const BasketExtra = (props: BasketExtraProps) => {
  const { service, onUndoRemoveAddOn, onRemoveAddOn, isBusiness, flags, planType, bundleType, packageId } = props
  const {
    skuId,
    displayName,
    priceDetails,
    name,
    displayDescription,
    headerStatus,
    description,
    productClass,
    packageLineId,
  }: ServiceValues = service

  const onUndo = () => {
    if (onUndoRemoveAddOn) {
      onUndoRemoveAddOn(packageId, skuId, 'SERVICE')
    }
  }

  const onRemove = () => {
    if (onRemoveAddOn) {
      onRemoveAddOn(packageId, packageLineId, skuId)
    }
  }

  const monthlyPrice = getPrice(priceDetails?.monthlyPrice, isBusiness)
  const upfrontPrice = getPrice(priceDetails?.oneOffPrice, isBusiness)
  const descriptionFromAllowances: React.ReactNode[] = getBundleDescription(service)
  const isEngineer: boolean = containsText([name, displayDescription, description], ['engineer', 'installation'])
  const isInsurance = isInsuranceService(service)
  const itemTitle: ServiceValues['displayName'] | ServiceValues['name'] = displayName || name
  let itemDescription: React.ReactNode[] = []
  if (displayDescription) itemDescription = [displayDescription]

  if (descriptionFromAllowances && descriptionFromAllowances.length) {
    itemDescription = descriptionFromAllowances
  }

  const isHBB = isBroadband(planType)
  const combinedName: string = (name || '' + displayName || '').toLowerCase()
  const isPaygAddon: boolean = combinedName.includes('big value bundle') || combinedName.includes('topup')
  const isBasicSimo: boolean = !!(planType?.toLowerCase() === 'simo' && bundleType?.toLowerCase().includes('basic'))
  const isBoltOn: boolean = productClass?.toLowerCase() === 'mbb traffic bolt-ons'
  const hasOnRemove = !isBasicSimo && !isPaygAddon && !isBoltOn && !isHBB && onRemoveAddOn
  const hasOnUndo = !isHBB && onUndoRemoveAddOn

  const basketItemProps = {
    key: skuId,
    title: itemTitle,
    description: itemDescription,
    iconAppearance: 'dark',
    onUndo: hasOnUndo ? onUndo : undefined,
    onRemove: hasOnRemove ? onRemove : undefined,
    upfrontPrice,
    monthlyPrice,
    headerStatus,
    isBusiness: isInsurance ? false : isBusiness,
    ...getPicture(isEngineer, isFixedLineService(service), isHBB, isInsurance, name),
    flags,
  }

  if (isHiddenService(service, isBusiness)) return null

  return <BasketItem {...basketItemProps} />
}

export default BasketExtra
