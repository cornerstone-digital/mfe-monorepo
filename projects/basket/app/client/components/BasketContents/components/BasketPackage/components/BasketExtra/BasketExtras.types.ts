import { OnUndoRemoveAddOn, OnRemoveAddOn, Flags } from '@pages/BasketPage/BasketPage.types'
import { ServiceValues } from '../PackageItemList/PackageItemList.types'

export interface BasketExtraProps {
  packageId: BasketV2.ModelPackage['packageId']
  service: ServiceValues
  planType: BasketV2.ModelPackage['planType']
  bundleType: BasketV2.Bundle['bundleType']
  isBusiness: boolean
  onUndoRemoveAddOn?: OnUndoRemoveAddOn
  onRemoveAddOn?: OnRemoveAddOn
  flags?: Flags
}

export interface MockBasketExtraData {
  packageId: BasketV2.ModelPackage['packageId']
  service: ServiceValues
  planType: BasketV2.ModelPackage['planType']
  isBroadband: boolean
}

export type BasketExtraPassImage = { [key: string]: string }
