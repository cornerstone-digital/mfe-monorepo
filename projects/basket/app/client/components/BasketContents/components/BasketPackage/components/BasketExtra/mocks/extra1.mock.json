{
  "packageId": "684b793c-88b6-4b3f-a4ab-863e26e37f75",
  "service": {
    "packageLineId": "6572d967-0aae-4cad-a032-e65860ec988b",
    "action": "ADD",
    "skuId": "110461",
    "name": "Vodafone Insurance: Tier 1 Loss Theft and Damage",
    "description": null,
    "displayName": "Loss, theft, damage and breakdown cover",
    "displayDescription": null,
    "specificationGroup": [],
    "merchandisingMedia": [],
    "attributes": [],
    "priceDetails": {
      "oneOffPrice": null,
      "oneOffDiscountPrice": null,
      "monthlyPrice": {
        "gross": "13.50",
        "net": "13.50",
        "vat": "0"
      },
      "monthlyDiscountPrice": {
        "gross": "13.50",
        "net": "13.50",
        "vat": "0"
      },
      "merchandisingPromotions": null
    },
    "freeGift": false,
    "productClass": "Insurance",
    "dataRollover": null,
    "rootProductId": "110461",
    "allowances": null,
    "rewardPoints": null,
    "planDuration": null,
    "isChanged": false,
    "headerStatus": "removed"
  },
  "planType": "HANDSET",
  "paymentType": "Postpaid"
}