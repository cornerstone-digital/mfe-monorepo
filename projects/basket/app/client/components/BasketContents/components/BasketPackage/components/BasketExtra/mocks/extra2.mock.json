{
  "packageId": "fee83b9d-d084-4ae3-a198-9ff630665159",
  "service": {
    "packageLineId": "17e76d94-44b7-4e00-935a-e2e0b283cdfa",
    "action": "ADD",
    "skuId": "110461",
    "name": "Vodafone Insurance: Tier 1 Loss Theft and Damage",
    "description": null,
    "displayName": "Loss, theft, damage and breakdown cover",
    "displayDescription": null,
    "specificationGroup": [],
    "merchandisingMedia": [],
    "attributes": [],
    "priceDetails": {
      "oneOffPrice": null,
      "oneOffDiscountPrice": null,
      "monthlyPrice": {
        "gross": "13.50",
        "net": "13.50",
        "vat": "0"
      },
      "monthlyDiscountPrice": {
        "gross": "13.50",
        "net": "13.50",
        "vat": "0"
      },
      "merchandisingPromotions": null
    },
    "freeGift": false,
    "productClass": "Insurance",
    "dataRollover": null,
    "rootProductId": "110461",
    "allowances": null,
    "rewardPoints": null,
    "planDuration": null,
    "isChanged": false
  },
  "planType": "HANDSET",
  "paymentType": "Postpaid"
}