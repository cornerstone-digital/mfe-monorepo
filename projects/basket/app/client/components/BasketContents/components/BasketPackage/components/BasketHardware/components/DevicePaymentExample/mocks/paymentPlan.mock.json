{
  "aprRepresentative": "0.0",
  "duration": {
    "uom": "MONTH",
    "value": "7"
  },
  "interestRate": "0.0",
  "description": "On a 7-month phone credit agreement",
  "monthlyPrice": {
    "uom": "£",
    "gross": "166.71",
    "net": "138.93",
    "vat": "27.78"
  },
  "totalHandsetCredit": {
    "uom": "£",
    "gross": "1167.00",
    "net": "972.50",
    "vat": "194.50"
  },
  "totalHandsetPrice": {
    "uom": "£",
    "gross": "1320.00",
    "net": "1100.00",
    "vat": "220.00"
  },
  "upFrontPrice": {
    "uom": "£",
    "gross": "153.00",
    "net": "127.50",
    "vat": "25.50"
  }
}