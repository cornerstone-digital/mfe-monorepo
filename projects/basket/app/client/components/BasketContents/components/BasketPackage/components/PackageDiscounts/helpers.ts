import { get } from 'lodash'

import { hasAccessoryDiscount } from '@helpers/typeCheck'

export const getPackageDiscountDescriptions = (packageItem: BasketV2.ModelPackage) => {
  const descriptions: string[] = []

  const discountDescriptions = packageItem?.bundle?.priceDetails?.listOfMerchandisingPromotion
    ?.map(promotion => promotion.label || '')
    .filter(item => !!item.length)

  if (discountDescriptions?.length) {
    descriptions.push(...discountDescriptions)
  }

  if (packageItem.hardwares?.length) {
    const hardwareDescriptions = packageItem.hardwares.reduce((items: string[], hardware: BasketV2.Hardware) => {
      if (hasAccessoryDiscount(hardware) && !!hardware.priceDetails?.merchandisingPromotions?.label) {
        items.push(hardware.priceDetails.merchandisingPromotions.label)
      }
      return items
    }, [])

    if (hardwareDescriptions.length) {
      descriptions.push(...hardwareDescriptions)
    }
  }

  return descriptions
}

const getPackageAccessoryDiscount = (packageItem: BasketV2.ModelPackage, priceType: 'net' | 'gross') => {
  return (
    packageItem.hardwares?.reduce((totalAmount, hardware) => {
      if (hasAccessoryDiscount(hardware)) {
        totalAmount += parseFloat(get(hardware, `priceDetails.oneOffPrice.${priceType}`) || '0')
        totalAmount -= parseFloat(get(hardware, `priceDetails.oneOffDiscountPrice.${priceType}`) || '0')
      }
      return totalAmount
    }, 0) || 0
  )
}

export const getPackageDiscountAmounts = (packageItem: BasketV2.ModelPackage, priceType: 'net' | 'gross') => {
  const totalMonthlySavingsAmount = parseFloat(get(packageItem, `bundle.priceDetails.totalMonthlySavingsAmount.${priceType}`) || '0')

  return {
    totalMonthlySavingsAmount,
    totalUpfrontDiscountAmount: getPackageAccessoryDiscount(packageItem, priceType),
  }
}
