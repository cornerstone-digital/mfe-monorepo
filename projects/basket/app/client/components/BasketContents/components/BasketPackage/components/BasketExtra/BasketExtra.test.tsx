import BasketExtra from './BasketExtra'
import mockExtra1 from './mocks/extra1.mock.json'
import mockExtra2 from './mocks/extra2.mock.json'
import transformUtils from '@shared/helpers/transformUtils/transformUtils'
import { MockBasketExtraData } from '@components/BasketContents/components/BasketPackage/components/BasketExtra/BasketExtras.types'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import * as getBundleDescription from '../../helpers/getBundleDescription'
import { mockStore } from '@tools/mocks/storeMock'

const handlers = {
  onRemoveAddOn: jest.fn(),
  onUndoRemoveAddOn: jest.fn(),
}

jest.mock('../../helpers/getBundleDescription', () => ({
  __esModule: true,
  default: jest.fn().mockReturnValue([]),
}))

jest.mock('@store', () => ({
  useStore: jest.fn().mockImplementation(() => mockStore({})),
}))

const mockExtraOne: MockBasketExtraData = { ...transformUtils.removeNulls(mockExtra1) }
const mockExtraTwo: MockBasketExtraData = { ...transformUtils.removeNulls(mockExtra2) }

describe('<BasketExtra />', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  const propsMock1 = {
    ...handlers,
    ...mockExtraOne,
    bundleType: '',
    isBusiness: false,
  }
  const propsMock2 = {
    ...handlers,
    ...mockExtraTwo,
    isBusiness: false,
    bundleType: '',
  }
  describe('renders', () => {
    it('mock1 correctly', () => {
      const { asFragment } = render(<BasketExtra {...propsMock1} />)
      expect(asFragment()).toMatchSnapshot()
    })
    it('mock2 correctly', () => {
      const { asFragment } = render(<BasketExtra {...propsMock2} />)
      expect(asFragment()).toMatchSnapshot()
    })
  })

  it('should not render if flexi upgrade', () => {
    const props = {
      ...propsMock1,
      planType: 'BROADBAND:FTTH',
      service: {
        ...propsMock1.service,
        productClass: 'fee',
        name: 'fee for flexi-upgrade',
      },
    }
    const { container } = render(<BasketExtra {...props} />)
    expect(container.textContent).toBe('')
  })

  it('should render description if provided', () => {
    const props = {
      ...propsMock1,
      service: {
        ...propsMock1.service,
        displayDescription: 'test description',
      },
    }
    const { getAllByText } = render(<BasketExtra {...props} />)
    expect(getAllByText(props.service.displayDescription)).toHaveLength(2)
  })

  it('should render description from allowances if provided in service', () => {
    const mockDescription = 'mock description'
    jest.spyOn(getBundleDescription, 'default').mockReturnValueOnce([mockDescription])
    const { getAllByText } = render(<BasketExtra {...propsMock1} />)
    expect(getAllByText(mockDescription)).toHaveLength(2)
  })

  it('should show undo button if package add on is undoable', () => {
    const { getByText } = render(<BasketExtra {...propsMock1} />)
    const undoButton = getByText('Undo').closest('button') as HTMLButtonElement
    expect(undoButton).toBeInTheDocument()

    userEvent.click(undoButton)
    expect(handlers.onUndoRemoveAddOn).toBeCalledWith(propsMock1.packageId, propsMock1.service.skuId, 'SERVICE')
  })

  it("should not show undo button is not present if it's a broadband plan", () => {
    const props = {
      ...propsMock1,
      planType: 'BROADBAND:FTTH',
    }
    const { queryByText } = render(<BasketExtra {...props} />)
    const undoButton = queryByText('Undo')
    expect(undoButton).not.toBeInTheDocument()
  })

  it("should not show remove button is not present if it's a simo basic plan", () => {
    const props = {
      ...propsMock1,
      planType: 'SIMO',
      bundleType: 'Basic',
    }
    const { queryByText } = render(<BasketExtra {...props} />)
    const removeButton = queryByText('Remove')
    expect(removeButton).not.toBeInTheDocument()
  })

  it('should show remove button if package add on is removable', () => {
    const { getByText } = render(<BasketExtra {...propsMock2} />)
    const removeButton = getByText('Remove').closest('button') as HTMLButtonElement
    expect(removeButton).toBeInTheDocument()

    userEvent.click(removeButton)
    expect(handlers.onRemoveAddOn).toBeCalledWith(propsMock2.packageId, propsMock2.service.packageLineId, propsMock2.service.skuId)
  })
})
