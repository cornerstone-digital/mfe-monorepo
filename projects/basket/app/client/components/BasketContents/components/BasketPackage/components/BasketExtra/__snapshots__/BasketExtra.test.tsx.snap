// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasketExtra /> renders mock1 correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBQB Gridstyle__GridContainer-sc-11onwkc-0 juDKGU bRCyJL no-gutters"
    data-name="container-fluid"
  >
    <div
      class="sc-gsTEea GridRowstyle__GridRow-sc-1l0dhq9-0 DvuGS dhUlIt"
      data-name="row"
    >
      <div
        class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 bVSFOy cRslqE header"
        data-name="col-12 col-md-12 col-lg-12"
      >
        <div
          stylename="item-header"
        >
          <div
            class="item-title"
          >
            <h5
              class="heading heading-6 inherit w-bold margin-0"
            >
              You removed Loss, theft, damage and breakdown cover
            </h5>
          </div>
          <button
            data-selector="basket-header-action-you-removed-loss-theft-damage-and-breakdown-cover"
            stylename="action-button"
          >
            <span
              class="action-text"
            >
              Undo
            </span>
            <span
              aria-hidden="true"
              class="wrapper inherit size-1 1  "
            >
              test-file-stub
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="item-body-wrap removed"
    >
      <div
        class="sc-gsTEea GridRowstyle__GridRow-sc-1l0dhq9-0 DvuGS dhUlIt item-body"
        data-name="row"
      >
        <div
          class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 heDFIj cRslqE icon-column-wrapper"
          data-name="col-4 col-md-2 col-lg-2"
        >
          <span
            aria-hidden="true"
            class="wrapper dark size-5 5  "
          >
            test-file-stub
          </span>
        </div>
        <div
          class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 fVIZqk cRslqE justify-left desc-column-large"
          data-name="col-12 col-md-5 col-lg-4"
        />
        <div
          class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 ebiDZn cRslqE"
          data-name="col-8 col-md-5 col-lg-6"
        >
          <div
            class="sc-bdfBQB Gridstyle__GridContainer-sc-11onwkc-0 juDKGU bRCyJL no-gutters"
            data-name="container-fluid"
          >
            <div
              class="sc-gsTEea GridRowstyle__GridRow-sc-1l0dhq9-0 DvuGS dhUlIt"
              data-name="row"
            >
              <div
                class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 dOTIbJ cRslqE column"
                data-name="col-6"
              >
                <div
                  class="basket-item-totals"
                  data-selector="cost-upfront"
                >
                  <p
                    class=""
                  >
                    <span
                      class="prefix inherit"
                    >
                      Upfront
                    </span>
                    <span
                      class="main inherit"
                    >
                      £0
                    </span>
                    <span
                      class="suffix inherit"
                    />
                    <span
                      class="suffix2 grey"
                    />
                  </p>
                </div>
              </div>
              <div
                class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 dOTIbJ cRslqE column"
                data-name="col-6"
              >
                <div
                  class="basket-item-totals"
                  data-selector="cost-monthly"
                >
                  <p
                    class=""
                  >
                    <span
                      class="prefix inherit"
                    >
                      Monthly
                    </span>
                    <span
                      class="main inherit"
                    >
                      £13.50
                    </span>
                    <span
                      class="suffix inherit"
                    />
                    <span
                      class="suffix2 grey"
                    />
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gsTEea GridRowstyle__GridRow-sc-1l0dhq9-0 DvuGS dhUlIt desc-column-small"
        data-name="row"
      >
        <div
          class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 fyWrjy cRslqE"
          data-name="col"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<BasketExtra /> renders mock2 correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBQB Gridstyle__GridContainer-sc-11onwkc-0 juDKGU bRCyJL no-gutters"
    data-name="container-fluid"
  >
    <div
      class="sc-gsTEea GridRowstyle__GridRow-sc-1l0dhq9-0 DvuGS dhUlIt"
      data-name="row"
    >
      <div
        class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 bVSFOy cRslqE header"
        data-name="col-12 col-md-12 col-lg-12"
      >
        <div
          stylename="item-header"
        >
          <div
            class="item-title"
          >
            <h5
              class="heading heading-6 inherit w-bold margin-0"
            >
              Loss, theft, damage and breakdown cover
            </h5>
          </div>
          <button
            data-selector="basket-header-action-loss-theft-damage-and-breakdown-cover"
            stylename="action-button"
          >
            <span
              class="action-text"
            >
              Remove
            </span>
            <span
              aria-hidden="true"
              class="wrapper inherit size-1 1  "
            >
              test-file-stub
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      aria-hidden="false"
      class="item-body-wrap"
    >
      <div
        class="sc-gsTEea GridRowstyle__GridRow-sc-1l0dhq9-0 DvuGS dhUlIt item-body"
        data-name="row"
      >
        <div
          class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 heDFIj cRslqE icon-column-wrapper"
          data-name="col-4 col-md-2 col-lg-2"
        >
          <span
            aria-hidden="true"
            class="wrapper dark size-5 5  "
          >
            test-file-stub
          </span>
        </div>
        <div
          class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 fVIZqk cRslqE justify-left desc-column-large"
          data-name="col-12 col-md-5 col-lg-4"
        />
        <div
          class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 ebiDZn cRslqE"
          data-name="col-8 col-md-5 col-lg-6"
        >
          <div
            class="sc-bdfBQB Gridstyle__GridContainer-sc-11onwkc-0 juDKGU bRCyJL no-gutters"
            data-name="container-fluid"
          >
            <div
              class="sc-gsTEea GridRowstyle__GridRow-sc-1l0dhq9-0 DvuGS dhUlIt"
              data-name="row"
            >
              <div
                class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 dOTIbJ cRslqE column"
                data-name="col-6"
              >
                <div
                  class="basket-item-totals"
                  data-selector="cost-upfront"
                >
                  <p
                    class=""
                  >
                    <span
                      class="prefix inherit"
                    >
                      Upfront
                    </span>
                    <span
                      class="main inherit"
                    >
                      £0
                    </span>
                    <span
                      class="suffix inherit"
                    />
                    <span
                      class="suffix2 grey"
                    />
                  </p>
                </div>
              </div>
              <div
                class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 dOTIbJ cRslqE column"
                data-name="col-6"
              >
                <div
                  class="basket-item-totals"
                  data-selector="cost-monthly"
                >
                  <p
                    class=""
                  >
                    <span
                      class="prefix inherit"
                    >
                      Monthly
                    </span>
                    <span
                      class="main inherit"
                    >
                      £13.50
                    </span>
                    <span
                      class="suffix inherit"
                    />
                    <span
                      class="suffix2 grey"
                    />
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gsTEea GridRowstyle__GridRow-sc-1l0dhq9-0 DvuGS dhUlIt desc-column-small"
        data-name="row"
      >
        <div
          class="sc-dlfnuX GridColumnstyle__GridColumn-sc-19afl1d-0 fyWrjy cRslqE"
          data-name="col"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
